src/app/shared/components/validation-hint/validation-hint.component.spec.ts

```diff
@@ -23,4 +24,84 @@ describe('ValidationHintComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+  
+  describe('tests of checkInvalidText method', () => {
+    const errorsCasesMap = [
+      ['invalidSymbols', 'NAME_REGEX', NAME_REGEX, true, false, false, false, false],
+      ['invalidCharacters', 'NO_LATIN_REGEX', NO_LATIN_REGEX, false, true, false, false, false],
+      ['invalidStreet', 'STREET_REGEX', STREET_REGEX, false, false, true, false, false],
+      ['invalidHouse', 'HOUSE_REGEX', HOUSE_REGEX, false, false, false, true, false],
+      ['invalidSectionName', 'SECTION_NAME_REGEX', SECTION_NAME_REGEX, false, false, false, false, true],
+    ];
+
+    test.each(errorsCasesMap)('should assign to %s TRUE if required pattern equal to %s',
+      (_, __, requiredPattern, invalidSymbols, invalidCharacters, invalidStreet, invalidHouse, invalidSectionName) => {
+        const errors = { pattern: { requiredPattern } };
+        component['checkInvalidText'](errors);
```

Comment:
Usually, in unit tests, we check only public methods

Reply:
So we don't have to write tests of private methods or it depends on situation and time that we have?

Reply:
Anyway, we use private methods elsewhere in public/protected methods.
If it requires much more effort, you can test a separate part of the logic that is in a private method.
Testing private methods is okay, but not preferred.